src/main/java/org/zalando/planb/revocation/util/MessageHasher.java

```diff
@@ -67,9 +68,8 @@ public String hashAndEncode(final RevocationType type, final String... messages)
         byte[] hashed = message.getBytes();
 
         if (hashingAlgorithms().containsKey(type)) {
-
-            hashingAlgorithms().get(type).update((salt() + message).getBytes());
-            hashed = hashingAlgorithms().get(type).digest();
+            String algorithm = hashingAlgorithms().get(type);
+            hashed = MessageDigest.getInstance(algorithm).digest((salt() + message).getBytes());
         }
 
         return Base64.getUrlEncoder().encodeToString(hashed);
```

Comment:
only thing is the what encoding is used here? For the string I mean? If it varies between nodes then the resulting byte array will be different.